zulipterminal/ui_tools/buttons.py

```diff
@@ -300,6 +300,11 @@ def handle_link(self, *_: Any) -> None:
         server_url = self.model.server_url
         if self.link.startswith(urljoin(server_url, '/#narrow/')):
             self.handle_narrow_link()
+        elif self.link.startswith(urljoin(server_url, '/user_uploads/')):
+            # Exit pop-up promptly, let the media download in the background.
+            if isinstance(self.controller.loop.widget, urwid.Overlay):
+                self.controller.exit_popup()
```

Comment:
What does this check avoid? I know you've used it elsewhere, but I don't remember what the conclusion was - can we discuss and then maybe add a comment. Does it belong in `exit_popup`?

Reply:
@preetmishra could you elaborate this?

Reply:
@Ezio-Sarthak The pop-ups open in an urwid Overlay. This check is to make sure `exit_popup` is only called when an Overlay exist.

Reply:
Ah I see that now. If I avoid that check, there is some unwanted synchronous behavior in footer and popup, if that makes any sense.